snippets/LimitedWordTextArea.md

```diff
@@ -0,0 +1,62 @@
+### LimitedWordTextArea
+
+Renders a textarea component with a word limit.
+
+Use the value of the `value` prop to determine the initial `state.content` and `state.wordCount` and the value of the `limit` props to determine the value of `state.limit`.
+Create a method, `handleChange`, which trims the `event.target.value` data if necessary and uses `Component.prototype.setState` to update `state.content` and `state.wordCount`, and bind it to the component's context.
+In the`render()` method, use a`<div>` to wrap both the`<textarea>` and the `<p>` element that displays the character count and bind the `onChange` event of the `<textarea>` to the `handleChange` method.
+
+```jsx
+class LimitedWordTextArea extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      content: props.value,
+      wordCount: props.value.split(' ').filter(Boolean).length,
```

Comment:
wouldn't we be able to put an initial value longer than the word limit?

Reply:
You're totally right. This is the only flaw as of now. Any idea on how to fix it?

Reply:
By creating the two methods. As long as you have a way of checking how long the string is and then trimming it it won't really matter. It's why I suggested adding in a couple of static helper functions